fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java

```diff
@@ -1863,7 +1864,9 @@ public void dropPartition(Database db, OlapTable olapTable, DropPartitionClause
         DropPartitionInfo info = new DropPartitionInfo(db.getId(), olapTable.getId(), partitionName, isTempPartition,
                 clause.isForceDrop(), recycleTime, version, versionTime);
         Env.getCurrentEnv().getEditLog().logDropPartition(info);
-
+        Env.getCurrentEnv().getEventProcessor().processEvent(
+                new DropPartitionEvent(db.getCatalog().getId(), db.getId(),
+                        olapTable.getId()));
```

Comment:
if we do not persist these events, what will happen if fe stop after logDropPartition and before processEvent?

Reply:
I changed the location where the event was generated to before writing the log, and waited for the message to be processed before writing the log. But here we don't care about the result of message processing (it's up to the specific listener to decide how to handle a failure)